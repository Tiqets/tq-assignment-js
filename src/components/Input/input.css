.input {
  height: 50px;
  font-size: 16px;
  width: 100%;
  border-radius: 5px;
  border: 1px solid #D6D8DC;
  padding: 0px 10px;
  color: #323C52;
}

.input::placeholder {
  color: #ADB1BA;
}