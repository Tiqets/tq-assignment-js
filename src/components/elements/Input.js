import React from 'react';

// Components
import InputGroup from '../InputGroup';
import Label from './Label';

const InputField = ({ name, placeholder, className, onChange }) => (
  <input
    type="text"
    id={`input-${name}`}
    name={name}
    className={className || 'form__cell-input'}
    placeholder={placeholder}
    onChange={onChange}
  />
  < />
);

const Input = props => <InputGroup {...props} component={InputField} />;

export default Input;
