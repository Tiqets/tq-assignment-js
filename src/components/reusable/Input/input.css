.input {
  height: 50px;
  font-size: 16px;
  width: 100%;
  border-radius: 5px;
  border: 1px solid #d6d8dc;
  padding: 0px 10px;
  color: #323c52;
}

.input::placeholder {
  color: #adb1ba;
}
