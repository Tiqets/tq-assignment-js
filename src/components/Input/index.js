import React from 'react';
import PropTypes from 'prop-types';
import './input.css';

/* Feel free to make this a functional component if you consider it necessary */
const Input = (props) => {
  const { placeholder } = props;
  return (
    <input className="input" type="text" placeholder={placeholder} />
  );
};

Input.propTypes = {
  placeholder: PropTypes.string,
};

Input.defaultProps = {
  placeholder: 'Input',
};

export default Input;
