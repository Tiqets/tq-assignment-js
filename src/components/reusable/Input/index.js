import React from 'react';
import PropTypes from 'prop-types';
import './input.css';

/* Feel free to make this a functional component if you consider it necessary */
const Input = (props) => {
  const { placeholder, name, onChangeHandler } = props;
  return (
    <input className="input" type="text" placeholder={placeholder} name={name} onChange={onChangeHandler} />
  );
};

Input.propTypes = {
  placeholder: PropTypes.string,
  name: PropTypes.string,
  onChangeHandler: PropTypes.func,
};

Input.defaultProps = {
  placeholder: 'Input',
  name: '',
  onChangeHandler: () => false,
};

export default Input;
