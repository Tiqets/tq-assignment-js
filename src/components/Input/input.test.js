/* global it, expect, document */
import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Input from './index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Input />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Handles the placeholder prop correctly', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Input placeholder="new placeholder" />, div);
  expect(div.getElementsByTagName('input')[0].getAttribute('placeholder')).toBe('new placeholder');
});

it('renders the input correctly with default props', () => {
  const TextInputComponent = TestRenderer.create(<Input />).toJSON();
  expect(TextInputComponent).toMatchSnapshot();
});
