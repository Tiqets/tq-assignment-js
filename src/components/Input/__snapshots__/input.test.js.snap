// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the input correctly with default props 1`] = `
<input
  className="input"
  placeholder="Input"
  type="text"
/>
`;
